new-cep.md

```diff
@@ -0,0 +1,204 @@
+    """
+    print("on_exception action")
+
+
+@hookimpl
+def conda_on_exception():
+    """
+    Returns our CondaOnException class which attaches our ``custom_plugin_on_exception_action``.
+    """
+    yield CondaOnException(
+        name=f"{PLUGIN_NAME}_on_exception",
+        action=custom_plugin_on_exception_action
+    )
+```
+
+
+## Motivation
+
+Our immediate motivations for adding these plugin hooks include the following:
+
+### Better authentication handling
+
+The `pre_command` hook will enable plugin authors to interrupt the normal start up of conda commands.
+For better authentication handling, this opens the door for either asking the user directly
+for their credentials or retrieving credentials from an OS keyring. These credentials can
+then be stored and used for the duration of the running command and subsequent runs too.
+
+A prototype of how this could function can be found in the following pull request:
+
+- https://github.com/conda/conda/pull/12139
+
+### Better exception logging
+
+The `on_exception` hook will enable plugin authors to provide a better experience for users and
+developers when exceptions are encountered. For example, a plugin could be written to display
+stacktraces in color or even drop you into a debugging console.
+
+### Plus more
+
+The above use cases are just a few possibilities out there. We know that our users and community will
+have many more ideas for how best to utilize these generic plugin hooks.
+
+## Rationale
+
+By introducing a set of generic hooks like the above, we grant plugin authors with quite a bit of
+flexibility for customizing how conda behaves. We believe starting out with a small set of generic
+plugin hooks is best so that we do not overwhelm would-be conda plugin authors. Depending on how much
+these plugin hooks are used and whether there is a demand, we may choose add more in the future with a 
+new CEP. For now though, we believe it is best to stick with this narrow selection as we slowly grow
+our plugin ecosystem.
+
+## FAQ
+
+- Will I have access to global variables such as `conda.base.context.context` in my plugin hook?
+    - Yes, for all plugin hooks listed in this CEP, they will be placed after the global context
+      object has been initialized.
+
+- Where exactly will the `pre_command` hooks be called?
+    - All registered `pre_command` hooks will be called before the `do_call` call in the
+      [`conda.cli.main:main_subshell`][conda-pre-invoke-location] function.
+
+- Where exactly will the `post_command` hooks be called?
+    - All registered `post_command` hooks will be called after the `do_call` call in the
+      [`conda.cli.main:main_subshell`][conda-pre-invoke-location] function.
+
+- Where exactly will the `on_exception` hooks be called?
+    - All registered `on_exception` hooks will be called inside the
+      [`conda.exceptions.ExceptionHandler`][conda-on-exception-location] class.
```

Comment:
I think the `conda.exceptions.ExceptionHandler` should be instead the default exception handler registered ([using the `tryfirst=True` hookimpl parameter](https://pluggy.readthedocs.io/en/stable/index.html?highlight=tryfirst#call-time-order))